src/cargo/core/resolver/mod.rs

```diff
@@ -451,7 +488,10 @@ fn activation_error(cx: &Context,
     let mut msg = msg;
     let all_req = semver::VersionReq::parse("*").unwrap();
     let new_dep = dep.clone().set_version_req(all_req);
-    let mut candidates = try!(registry.query(&new_dep));
+    let mut candidates: Vec<Summary> = match registry.query(&new_dep) {
+        Ok(candidates) => candidates,
+        Err(e) => return e,
+    };
```

Comment:
It can't trivially because I changed the return type of `activation_error` from `CargoResult<Box<Context>>` to `Box<CargoError>` (since it never actually returned a `Context`). I could change that back, but I'd need to explicitly unwrap `last_err` in `find_candidate`, and it just seems odd to offer an `Ok` result that can never actually happen.